### PR TITLE
Flash: Gossip latest update alongside the Channel config

### LINKED ISSUE #1910: Flash: Channel updates are ignored unless channel open message was received first
There is a race condition where channelOpen and channelUpdate messages are sent at roughly the same time. Any node which receives them out of order will currently reject the channel update message.

The only reason this doesn't happen in the tests is because most of the APIs were blocking and therefore assumed a certain order of messages.

### PR DESCRIPTION
**Pull request by omerfirmak**
Fixes #1910

### CODE REVIEW

File: source/agora/flash/api/FlashAPI.d [hunk 1]
```diff
@@ -29,6 +29,12 @@ import vibe.data.serialization;
 import vibe.http.common;
 import vibe.web.rest;
 
+struct ChannelOpen
```
Comment (Geod24) [inline source/agora/flash/api/FlashAPI.d]:
```suggestion
public struct ChannelOpen
```

Also this is public API, so documentation on the struct and all members plz

File: source/agora/flash/api/FlashAPI.d [hunk 2]
```diff
@@ -29,6 +29,12 @@ import vibe.data.serialization;
...
+{
+    ChannelConfig conf;
+    ChannelUpdate update;
+};
```
Comment (Geod24) [inline source/agora/flash/api/FlashAPI.d]:
```suggestion
}
```

File: source/agora/flash/api/FlashAPI.d [hunk 3]
```diff
@@ -29,6 +29,12 @@ import vibe.data.serialization;
 import vibe.http.common;
 import vibe.web.rest;
 
+struct ChannelOpen
+{
+    ChannelConfig conf;
+    ChannelUpdate update;
```
Comment (Geod24) [inline source/agora/flash/api/FlashAPI.d]: `public` + ddoc

**Review by Geod24 — APPROVED** [review]
LGTM, just a tiny style nit, then feel free to self-merge

**Comment by omerfirmak** [comment]
```
node-7_1  | 2021-08-30T08:52:40.724258704Z core.exception.AssertError@source/agora/consensus/protocol/Nominator.d(1126): utxo is Hash.init
node-7_1  | 2021-08-30T08:52:40.817886727Z ----------------
node-7_1  | 2021-08-30T08:52:40.817898027Z source/agora/node/main.d:276 nothrow void agora.node.main.handleAssertion(immutable(char)[], ulong, immutable(char)[]) [0x5558015a8230]
node-7_1  | 2021-08-30T08:52:40.817901327Z source/agora/consensus/protocol/Nominator.d:1106 _ZN9Nominator11getNodeUTXOEmm [0x555801633a90]
node-7_1  | 2021-08-30T08:52:40.817904127Z source/agora/consensus/protocol/Nominator.d:575 @trusted void agora.consensus.protocol.Nominator.Nominator.handleSCPEnvelope(in ref scpd.types.Stellar_SCP.SCPEnvelope) [0x555801631c50]
node-7_1  | 2021-08-30T08:52:40.817907027Z source/agora/consensus/protocol/Nominator.d:218 void agora.consensus.protocol.Nominator.Nominator.envelopeProcessTask() [0x555801631b10]
node-7_1  | 2021-08-30T08:52:40.817909727Z submodules/vibe-core/source/vibe/core/core.d:1212 nothrow @safe void vibe.core.core.setTimer(core.time.Duration, void delegate(), bool).__lambda4() [0x555801dc8a30]
node-7_1  | 2021-08-30T08:52:40.817912627Z submodules/vibe-core/source/vibe/core/core.d:1270 nothrow @safe void vibe.core.core.createTimer(void delegate() nothrow @safe).C.opCall(vibe.core.core.Timer).__lambda2(vibe.core.core.Timer) [0x555801dc8c00]
node-7_1  | 2021-08-30T08:52:40.817920227Z submodules/vibe-core/source/vibe/core/task.d:708 nothrow void vibe.core.task.TaskFuncInfo.set!(void delegate(vibe.core.core.Timer) nothrow @safe, vibe.core.core.Timer).set(ref void delegate(vibe.core.core.Timer) nothrow @safe, ref vibe.core.core.Timer).callDelegate(ref vibe.core.task.TaskFuncInfo) [0x555801e13450]
node-7_1  | 2021-08-30T08:52:40.817923727Z submodules/vibe-core/source/vibe/core/task.d:737 void vibe.core.task.TaskFuncInfo.call() [0x555801e10620]
node-7_1  | 2021-08-30T08:52:40.817926328Z submodules/vibe-core/source/vibe/core/task.d:403 nothrow void vibe.core.task.TaskFiber.run() [0x555801e0fbe0]
node-7_1  | 2021-08-30T08:52:40.817989429Z ??:? fiber_entryPoint [0x7fce18f4e0a0]
```

**Comment by Geod24** [comment]
CC @linked0

**Comment by linked0** [comment]
> ```
> node-7_1  | 2021-08-30T08:52:40.724258704Z core.exception.AssertError@source/agora/consensus/protocol/Nominator.d(1126): utxo is Hash.init
> node-7_1  | 2021-08-30T08:52:40.817886727Z ----------------
> ```

These lines can be removed. 
https://github.com/bosagora/agora/blob/bac5a52ecd281826f4e7b023c34d4b8b1241a6d5/source/agora/consensus/protocol/Nominator.d#L1123-L1127
@omerfirmak Can you try it? I will look into it more later.

**Pull request merged**